src/styles/blocks/_calendar.scss

```diff
@@ -0,0 +1,38 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $day-gap;
+  width: $calendar-size;
+  padding: $calendar-padding;
+  background-color: #fff;
+
+  &__day {
+    @extend %center;
+    gap: $day-gap;
+    width: $day-size;
+    height: $day-size;
+    font-size: 30px;
+    font-family: Arial, Helvetica, sans-serif;
+    background-color: #eee;
+    border: 1px solid black;
+    transition: all 0.5s;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: " #{$i}";
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translateY(-20px);
+    }
+  }
+
+  @each $day, $index in $days-col {
+    &__start--day-#{$day} :first-child {
+      margin-left: $index * ($day-size + $day-gap) - $day-size - $day-gap;
```

Comment:
how about `($index - 1) * ($day-size + $day-gap)` Then you can go beyond that and think about another improvement regarding `$index`